plutus-ledger/src/Ledger/Tx/Internal.hs

```diff
@@ -172,6 +187,49 @@ scriptTxInputs = (\x -> folding x) . filter $ \case
 --     _                                   -> Nothing
 -- inScripts _ _ = Nothing
 
+newtype TxOut = TxOut {getTxOut :: C.TxOut C.CtxTx C.BabbageEra}
+    deriving stock (Show, Eq, Generic)
+    deriving anyclass (ToJSON, FromJSON)
+
+instance C.ToCBOR TxOut where
+  toCBOR (TxOut txout) = C.toCBOR $ C.toShelleyTxOut C.ShelleyBasedEraBabbage txout
+
+instance C.FromCBOR TxOut where
+  fromCBOR = do
+    txout <- C.fromCBOR
+    pure $ TxOut $ C.fromShelleyTxOut C.ShelleyBasedEraBabbage txout
+
+instance Serialise TxOut where
+  encode = C.toCBOR
+  decode = C.fromCBOR
+
+instance NFData TxOut where
+  rnf (TxOut tx) = seq tx ()
+
+instance OpenApi.ToSchema TxOut where
+    declareNamedSchema _ = do
+      addressSchema <- OpenApi.declareSchemaRef (Proxy :: Proxy (C.AddressInEra C.BabbageEra))
+      valueSchema <- OpenApi.declareSchemaRef (Proxy :: Proxy Value)
+      bsSchema <- OpenApi.declareSchemaRef (Proxy :: Proxy Datum)
+      pure $ OpenApi.NamedSchema (Just "TxOut") $ mempty
+        & OpenApi.type_ ?~ OpenApi.OpenApiObject
+        & OpenApi.properties .~
+          [ ("address", addressSchema)
+          , ("value", valueSchema)
+          , ("datum", bsSchema)
+          , ("referenceScript", bsSchema)
+          ]
+        & OpenApi.required .~ ["address","value"]
+
+
+instance Pretty TxOut where
+  pretty (TxOut (C.TxOut addr v dh _rs)) =
+    hang 2 $ vsep
+      ["-" <+> pretty (fromCardanoTxOutValue v) <+> "with datum"
+      , pretty (fromCardanoTxOutDatumHash dh) <+> "addressed to"
+      , pretty (fromCardanoAddressInEra addr)
+      ]
```

Comment:
You need to handle the empty cases and reference scripts.

We would want the following possible outputs

```
- Value [...] addressed to
  ScriptCredential ....
  with datum hash .........
  and with reference script hash ......
```

or 

```
- Value [...] addressed to
  ScriptCredential ....
  with no datum
  and with no reference script
```